modules/querying/pages/query-optimizer/index.adoc

```diff
@@ -0,0 +1,34 @@
+= Query Optimizer
+:description: Overview of the query optimizer.
+
+GSQL has a built-in query optimizer that can optimize the execution plan of a query when you install the query.
+Currently, the optimizer can optimize the execution plan in the following ways:
+
+* <<_cost_based_optimization>>
+
+== What is the query optimizer?
+The query optimizer optimizes the compiled binary execution plan of a query.
+
+When you xref:query-operations.adoc#_install_query[install a GSQL query], GSQL compiles the query logic into a binary execution plan.
+If your query contains a path pattern, the pattern contains a source vertex set, one mor more path edge patterns, and one or more step vertex sets.
+Similar to the path pattern in GSQL, its binary execution plan also starts from a set of source vertices, traverses through edges in the specified pattern, and finds the target vertices.
+
+However, the actual execution plan of the query might perform path traversals differently than specified in the GSQL query to product results.
+When you install a query, the query optimizer can optimize the execution plan of a query so that it executes more efficiently.
+
+Depending on the type of optimization used, this could allow a query to execute with less time, less memory, or less storage.
+
+[#_cost_based_optimization]
+== Cost-based optimization
+
+Cost-based optimization reduces the number of edge traversals a query has to execute.
+Enabling the cost-based optimizer when installing a query has the potential to significantly reduce the execution time of the installed query.
+
+=== How it works
+The cost-based optimizer takes statistics regarding the vertex and edge types as well as their attributes that are referenced in the query to compile an execution plan that involves the fewest edge traversals.
```

Comment:
It's true that right now we're mostly trying to minimize the number of edge traversals, but with time our cost function will have more parameters. Do we want to call this out explicitly in the documentation? What do you think @ainurt?

Reply:
at least for now we can probably omit the details of our cost-model ?!

Reply:
The reason I felt the need to call this out is that as a user, this was the only tangible way i was able to understand what "cost" means. Cost could mean many different things to people - people could understand it as memory or other resources since it is a broad term in its daily usage. 

However, if we feel strongly that this is a confidential detail that we don't want to disclose, I can probably think of another way to explain it.